db/people.go

```diff
@@ -22,3 +26,25 @@ func UpdateUserProperties(ctx *api.Context, userID string, newProps api.PersonPr
 func UpdateIdentityState(ctx *api.Context, id, state string) error {
 	return ctx.DB().Table("identities").Where("id = ?", id).Update("state", state).Error
 }
+
+func CreatePerson(ctx *api.Context, username, hashedPassword string) (*models.Person, error) {
+	tx := ctx.DB().Begin()
+	defer tx.Rollback()
+
+	person := models.Person{Name: username, Type: "agent"}
+	if err := tx.Clauses(clause.Returning{}).Create(&person).Error; err != nil {
+		return nil, err
+	}
+
+	accessToken := models.AccessToken{
```

Comment:
I would wrap this in a db.CreateAccessToken(person, name, expiry) that generates, then saves the token returning the plain text token